packages/react/src/internal/components/InputLabel.tsx

```diff
@@ -52,10 +56,10 @@ const InputLabel: React.FC<React.PropsWithChildren<Props>> = ({
       }}
       {...props}
     >
-      {required ? (
+      {required || requiredText ? (
         <Box display="flex" as="span">
           <Box mr={1}>{children}</Box>
-          <span>*</span>
+          <span aria-hidden={!requiredIndicator ? true : undefined}>{!requiredText ? '*' : requiredText}</span>
```

Comment:
Personal preference for inverting the input variables here to remove the `!` (and maybe using the Nullish coalescing operator 🤔)

```suggestion
          <span aria-hidden={requiredIndicator ? undefined : true}>{requiredText ?? '*'}</span>
```

Bit of a nitpick so will approve anyway ✅